src/test/java/pl/jakubkozlowski/learning/firststeps/controller/ChampionControllerTest.java

```diff
@@ -0,0 +1,123 @@
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ChampionController.class)
+public class ChampionControllerTest {
+
+
+    @MockBean
+    public ChampionService championService;
+
+    @Autowired
+    private MockMvc mvc;
+
+    private ObjectMapper objectMapper;
+
+    private ChampionDTO championDTOAatrox;
+    private ChampionDTO championDTOAhri;
+    private ChampionDTO championDTOAnivia;
+
+    private List<ChampionDTO> championDTOList;
+
+    @Before
+    public void setUp() throws Exception {
+
+        championDTOAatrox = new ChampionDTO(ID_1, AATROX);
+        championDTOAhri = new ChampionDTO(ID_2, AHRI);
+        championDTOAnivia = new ChampionDTO(ID_3, ANIVIA);
+
+        championDTOList = new ArrayList<>(3);
+        championDTOList = Arrays.asList(championDTOAatrox, championDTOAhri, championDTOAnivia);
+
+        objectMapper = new ObjectMapper();
+
+        Mockito.when(championService.findAll()).thenReturn(championDTOList);
+        Mockito.when(championService.findById(ID_1)).thenReturn(championDTOAatrox);
+
+    }
+
+    @Test
+    public void whenFindAll_thenReturnJsonArrayOfChampions() throws Exception {
+
+
+        String content = mvc.perform(get("/champions")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        List<ChampionDTO> actualList = objectMapper.readValue(content, new TypeReference<List<ChampionDTO>>() {
+        });
+        assertThat(actualList).isEqualTo(championDTOList);
+    }
+
+    @Test
+    public void whenFindOne_thenReturnJsonArrayOfChampionWithGivenID() throws Exception {
+
+        String content = mvc.perform(get("/champions/" + ID_1)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+
+        ChampionDTO actual= objectMapper.readValue(content, new TypeReference<ChampionDTO>(){});
+        assertThat(actual).isEqualTo(championDTOAatrox);
+    }
+
+    @Test
+    public void whenPersist_thenReturnJsonArrayOfPersistChampion() throws Exception {
+
+        String content=mvc.perform(post("/champions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsBytes(championDTOAatrox)))
+                .andExpect(status().isCreated())
+                .andReturn().getResponse().getContentAsString();
+
+        ChampionDTO actual= objectMapper.readValue(content, new TypeReference<ChampionDTO>(){});
+        assertThat(actual).isEqualTo(championDTOAatrox);
+
+    }
+
+    @Test
+    public void whenUpdate_thenReturnJsonArrayOfPersistChampion() throws Exception {
+        String content=mvc.perform(put("/champions/"+ID_1)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsBytes(championDTOAatrox)))
```

Comment:
it would probably be easier in case of debugging need to use writeValueAsString to get JSON format which is also acceptable